libcontainer/specconv/spec_linux.go

```diff
@@ -333,7 +333,10 @@ func CreateLibcontainerConfig(opts *CreateOpts) (*configs.Config, error) {
 
 func createLibcontainerMount(cwd string, m specs.Mount) (*configs.Mount, error) {
 	if !filepath.IsAbs(m.Destination) {
-		return nil, fmt.Errorf("mount destination %s not absolute", m.Destination)
+		// Relax validation for backward compatibility
+		oldDest := m.Destination
+		m.Destination = filepath.Join("/", m.Destination)
```

Comment:
Hm... actually; did it previously take `cwd` into account? i.e. does this have to be;

```suggestion
		m.Destination = filepath.Join("/", cwd, m.Destination)
```

Reply:
Yes, non-absolute paths are relative to CWD (and this is what makes them kinda dangerous -- thus the requirement in OCI spec for paths to be absolute I guess), and so `filepath.Join("/", ...)` is the same as `filepath.Abs()` with CWD=="/").

I like filepath.Abs() better despite it being dependent on cwd -- this is how all relative paths should be treated.

Reply:
I can change it back to `filepath.Abs()`

Hmmm.... but checking the old behaviour; looks like it does not take the working directory of the container into account, so paths were treated as "relative to `/` (inside the container)"

```bash
docker run --rm -w /somepath -v hello busybox ls -la
total 8
drwxr-xr-x    2 root     root          4096 Jun  7 21:56 .
drwxr-xr-x    1 root     root          4096 Jun  7 21:56 ..

docker run --rm -w /somepath -v hello busybox ls -la /
total 52
drwxr-xr-x    1 root     root          4096 Jun  7 21:56 .
drwxr-xr-x    1 root     root          4096 Jun  7 21:56 ..
-rwxr-xr-x    1 root     root             0 Jun  7 21:56 .dockerenv
drwxr-xr-x    2 root     root         12288 Apr  7 19:47 bin
drwxr-xr-x    5 root     root           340 Jun  7 21:56 dev
drwxr-xr-x    1 root     root          4096 Jun  7 21:56 etc
drwxr-xr-x    2 root     root          4096 Jun  7 21:56 hello
drwxr-xr-x    2 nobody   nobody        4096 Apr  7 19:47 home
dr-xr-xr-x  201 root     root             0 Jun  7 21:56 proc
drwx------    2 root     root          4096 Apr  7 19:47 root
drwxr-xr-x    2 root     root          4096 Jun  7 21:56 somepath
dr-xr-xr-x   13 root     root             0 Jun  7 21:56 sys
drwxrwxrwt    2 root     root          4096 Apr  7 19:47 tmp
drwxr-xr-x    3 root     root          4096 Apr  7 19:47 usr
drwxr-xr-x    4 root     root          4096 Apr  7 19:47 var
```

Reply:
Tried a build with `filepath.Abs`, and looks like that's giving the same result; let me know if you like that more (and if you want me to squash)

Reply:
You are right -- the non-absolute paths are treated as they are relative to the container root (as in `securejoin.SecureJoin(rootfs, m.Destination)`, where `rootfs` is container root, and `m.Destination` is what supposed to be an absolute path).

What I think we should do is
```diff
-                return nil, fmt.Errorf("mount destination %s not absolute", m.Destination)
+               // TODO (runc 1.2.0): replace a warning below with
+               // return nil, fmt.Errorf("mount destination %s not absolute", m.Destination)
+               logrus.Warnf("mount destination %s not absolute (this won't be supported past runc 1.2)", m.Destination)
```

and use the m.Destination as-is.

Reply:
In general it's a dark territory as to when we'll release runc 1.2.0, but I guess at least now for a year from now.

Reply:
> What I think we should do is

👍 on adding the comment, and yes, removing in 1.1.0 or 1.2.0 SGTM (hoping that users by then have upgraded to more current runtimes)

> and use the m.Destination as-is.

SGTM. I was thinking that migrating it in code would help future upgrades, but I guess the actual config for these containers would be managed by higher level runtimes, so not an issue.

Reply:
Updated with your suggestion